include/aspect/compat.h

```diff
@@ -23,6 +23,26 @@
 
 #include <aspect/global.h>
 
+#if DEAL_II_VERSION_GTE(8,4,0)
+#else
+template <int dim>
+dealii::Tensor<1,dim> cross_product(const dealii::Tensor<1,dim> &a, const dealii::Tensor<1,dim> &b)
```

Comment:
I tend to think that you may be happied to actually package this into a `namespace dealii {...}` construct to ensure that name lookup works as it did before. I don't know whether it's going to make a difference in practice, but it may be a more straightforward way than qualifying everything this way.